agent/util-scripts/pbench-sysinfo-dump

```diff
@@ -116,10 +120,18 @@ function collect_sos {
 		# ref for plugin tuned: https://github.com/sosreport/sos/commit/c55c58cf2dabf93c924c839c8ed045c18e31ba1c
 		_modules="${_modules} tuned"
 	fi
+	if [[ "${sos_ver}" -ge 4 ]]; then
+		# the command is now "sos report" in 4.0+
+		_sos_report_cmd="sos report"
+	fi
+	if [[ "${sos_ver}" -gt 4 || ( "${sos_ver}" -eq 4 && "${sos_ver_minor}" -ge 1 )]]; then
+		# sos uses sha256 sums, not md5, in 4.1+
+		_cksum_suffix="sha256"
+	fi
 
 	_name="pbench-$(hostname -f)"
 	_cmd="${dir}/sosreport-${_name}.cmd"
-	printf -- "sosreport" > ${_cmd}
+	printf -- "%s" "${_sos_report_cmd}" > ${_cmd}
```

Comment:
While this obviously works, it's more complex than it needs to be; e.g.:
```
	printf -- "${_sos_report_cmd}" > ${_cmd}
```
If you're worried about there being percent signs in the command string, then use something even simpler, like:
```
	echo ${_sos_report_cmd} > ${_cmd}
```

Reply:
See my reply in #2372.